### ISSUE TITLE
Finalized and Safe Height Metrics are Zero (0) on Prometheus Endpoint

### ISSUE THREAD

**Issue by ncavedale-xlabs** [maticnetwork/bor#1388]
Hi team!
Metrics `chain_head_finalized` and `chain_head_safe` return "0" for `Bor`:

```
# TYPE chain_head_finalized gauge
chain_head_finalized 0

# TYPE chain_head_safe gauge
chain_head_safe 0
```

Is this by design? Are there plans to implement/fix these?

Thanks!

**Comment by marcello33** [comment]
Hey @ncavedale-xlabs, what version of bor are you using?

**Comment by MqllR** [comment]
After looking at the bor codebase, it seems like the finality isn't handled exactly the same way as geth. When you run a `eth_getBlockByNumber` with the finalize tag, you reach some bor specific code that uses the milestones https://github.com/maticnetwork/bor/blob/master/eth/api.go#L56.
The metric [chain_milestone_latest](https://github.com/maticnetwork/bor/blob/master/eth/downloader/whitelist/milestone.go#L38) seems to be pretty close to the block number retrieved with a `eth_getBlockByNumber` call. But I let the Polygon experts confirm or not :)

**Comment by avalkov** [comment]
Yes, finality is not handled the same way. To get final block you must get latest milestone or checkpoint.
